RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -97,16 +97,47 @@ class ManageSubscriptionsViewModel: ObservableObject {
         let customerInfo = try await purchasesProvider.customerInfo()
 
-        guard let currentEntitlement = customerInfo.earliestExpiringAppStoreEntitlement(),
-              let subscribedProduct = await purchasesProvider.products([currentEntitlement.productIdentifier]).first
+        // Find earliest expiring subscription
+        guard let earliestProductId =
+                customerInfo.activeSubscriptions.compactMap({ productId -> (String, Date)? in
+                    guard let expirationDate = customerInfo.expirationDate(forProductIdentifier: productId) else { return nil }
+                    return (productId, expirationDate)
+                })
+            .sorted(by: { $0.1 < $1.1 })
+            .first?
+            .0
         else {
+            // If no subscriptions found, check for lifetime purchases
+            if let firstNonSub = customerInfo.nonSubscriptions.first,
+               let entitlement = customerInfo.entitlements.active.values.first,
+               let product = await purchasesProvider.products([firstNonSub.productIdentifier]).first {
+                let subscriptionInformation = SubscriptionInformation(entitlement: entitlement,
+                                                                      subscribedProduct: product)
+                self.subscriptionInformation = subscriptionInformation
+                return
+            }
             Logger.warning(Strings.could_not_find_subscription_information)
             throw CustomerCenterError.couldNotFindSubscriptionInformation
         }
 
-        let subscriptionInformation = SubscriptionInformation(entitlement: currentEntitlement,
-                                                              subscribedProduct: subscribedProduct)
-        self.subscriptionInformation = subscriptionInformation
+        guard let product = await purchasesProvider.products([earliestProductId]).first else {
+            Logger.warning(Strings.could_not_find_subscription_information)
+            throw CustomerCenterError.couldNotFindSubscriptionInformation
+        }
+
+        // If we find a matching entitlement, use it. Otherwise, just use the product
+        if let entitlement = customerInfo.entitlements.active.values.first(where: {
+            $0.productIdentifier == earliestProductId
+        }) {
+            let subscriptionInformation = SubscriptionInformation(entitlement: entitlement,
+                                                                  subscribedProduct: product)
+            self.subscriptionInformation = subscriptionInformation
+        } else {
+            let expirationDate = customerInfo.expirationDate(forProductIdentifier: product.productIdentifier)
+            let subscriptionInformation = SubscriptionInformation(product: product,
+                                                                  expirationDate: expirationDate)
+            self.subscriptionInformation = subscriptionInformation
```

Comment:
I am planning on directly loading SubscriptionInformation in `CustomerCenterViewModel` to make this easier to understand and less error prone. But that comes in a separate PR